### PR TITLE
[TIMOB-24582] Android: Implement Ti.UI.NavigationWindow

### PR DESCRIPTION
**Pull request by garymathews**
- Implement `Titanium.UI.NavigationWindow`

###### TEST CASE
```JS
var win2 = Titanium.UI.createWindow({
    backgroundColor: 'red',
    title: 'Red Window'
});

var win1 = Titanium.UI.createNavigationWindow({
    window: win2
});

var win3 = Titanium.UI.createWindow({
    backgroundColor: 'yellow',
    title: 'Yellow Window'
});

var win4 = Titanium.UI.createWindow({
    backgroundColor: 'green',
    title: 'Green Window'
});

var button = Titanium.UI.createButton({
    title: 'Open Yellow Window'
});
button.addEventListener('click', function () {
    win1.openWindow(win3);
});

win2.add(button);
var button2 = Titanium.UI.createButton({
    title: 'Open Green Window'
});
button2.addEventListener('click', function () {
    win1.openWindow(win4);
});

win3.add(button2);
win1.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-24582)

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
@garymathews If this is intended for 8.0.0, please rebase on `next` branch and target that front he PR here. I tried changing to target `next` here, but it then picked up a lot of extra commits.

**Comment by keerthi1032** [comment]
FR Passed. Ti.UI.NavigationWindow works as expected.Waiting for CR to merge.
Name                        = Mac OS X
  Version                     = 10.13.6
  Architecture                = 64bit
  Memory                      = 17179869184
Node.js
  Node.js Version             = 8.12.0
  npm Version                 = 6.4.1
Titanium CLI
  CLI Version                 = 5.1.1
Titanium SDK
  SDK Version                 = local  8.0.0.v20181109091107 
Device                            =Pixel android 9
Emulator                        =Pixel android 8,Nexus 5

File: android/runtime/common/src/js/titanium.js [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
  * Appcelerator Titanium Mobile
-
- * Copyright (c) 2011-2013 by Appcelerator, Inc. All Rights Reserved.
+ *
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
  * Licensed under the terms of the Apache Public License
  * Please see the LICENSE included with this distribution for details.
  */
+/* global kroll */
+'use strict';
```
Comment (sgtcoolguy) [inline android/runtime/common/src/js/titanium.js]: @garymathews So I think adding this line actually is causing the build failure where it's complaining about line 77 here trying to assign a value to `this.currentActivity`. Toggling this off locally does "fix" it.

Which means this has been silently failing forever. Do we still need these wrapper types?

Do we need to make the property readwrite? It's defined as read-only in our proxy here: https://github.com/appcelerator/titanium_mobile/blob/master/android/modules/android/src/java/ti/modules/titanium/android/AndroidModule.java#L599

**Comment by sgtcoolguy** [comment]
Manually merged

**Pull request closed without merging**